src/distribution/Airdrop.sol

```diff
@@ -0,0 +1,78 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+pragma solidity 0.8.24;
+
+import { IERC20, SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import { Ownable } from "@openzeppelin/contracts/access/Ownable.sol";
+import { MerkleProof } from "@openzeppelin/contracts/utils/cryptography/MerkleProof.sol";
+import { EnumerableSet } from "@openzeppelin/contracts/utils/structs/EnumerableSet.sol";
+
+import { IAirdrop } from "./interfaces/IAirdrop.sol";
+
+/// @title Airdrop
+/// @notice A contract for distributing token airdrops using a merkle tree
+/// @dev Inspired by Uniswap https://github.com/Uniswap/merkle-distributor/blob/master/contracts/MerkleDistributor.sol at commit 25a79e8
+contract Airdrop is IAirdrop, Ownable {
```

Comment:
This can be done in a follow up if you prefer, but let's incorporate root types into this:
- `Token18 token`
- `UFixed18 amount`